Formula/apache-httpd.rb

```diff
@@ -0,0 +1,128 @@
+  sha256 "71fcc128238a690515bd8174d5330a5309161ef314a326ae45c7c15ed139c13a"
+
+  bottle do
+    sha256 "d35e67ae745053cda23273a520ead6bde506ec8082057edcd3185c9e36eae483" => :sierra
+    sha256 "4ea7056d6c08f9cf49507869aa8c78e3522c601d87b3347d31561ba05b33d41c" => :el_capitan
+    sha256 "e6ae5c4d38b40bcf3d85877ac117a1ea048de774edec3c2b0ff714bc01a6d40a" => :yosemite
+  end
+
+  deprecated_option "with-http2" => "with-nghttp2"
+
+  depends_on "apr"
+  depends_on "apr-util"
+  depends_on "openssl"
+  depends_on "pcre"
+
+  depends_on "nghttp2" => :optional
+
+  def install
+    # point config files to opt_prefix instead of the version-specific prefix
+    inreplace "Makefile.in",
+      '#@@ServerRoot@@#$(prefix)#', '#@@ServerRoot@@'"##{opt_prefix}#"
+
+    args = %W[
+      --enable-mods-shared=all
+      --enable-unique-id
+      --enable-ssl
+      --enable-dav
+      --enable-cache
+      --enable-logio
+      --enable-deflate
+      --enable-cgi
+      --enable-cgid
+      --enable-suexec
+      --enable-rewrite
+      --with-apr=#{Formula["apr"].opt_prefix}
+      --with-apr-util=#{Formula["apr-util"].opt_prefix}
+      --with-pcre=#{Formula["pcre"].opt_prefix}
+      --with-ssl=#{Formula["openssl"].opt_prefix}
+      --with-port=8080
+      --with-sslport=8443
+      --with-mpm=event
+      --prefix=#{prefix}
+      --mandir=#{man}
+      --sysconfdir=#{etc}/apache2
+      --datadir=#{var}/www
+      --localstatedir=#{var}/apache2
+    ]
+
+    if build.with? "nghttp2"
+      args << "--enable-http2"
+      args << "--with-nghttp2=#{Formula["nghttp2"].opt_prefix}"
+    end
+
+    (etc/"apache2").mkpath
+
+    system "./configure", *args
+    system "make", "install"
+
+    # remove non-executable files in sbin dir (for brew audit)
+    rm bin/"envvars"
+
+    unless (var/"log/apache2").exist?
+      (var/"log/apache2").mkpath
+      touch "#{var}/log/apache2/access_log"
+      touch "#{var}/log/apache2/error_log"
+    end
+  end
+
+  def post_install
+    (var/"run/apache-httpd").mkpath
+  end
+
+  plist_options :manual => "apachectl start"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/httpd</string>
+        <string>-D</string>
+        <string>FOREGROUND</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    begin
+      (testpath/"httpd.conf").write <<-EOS.undent
+        ServerName localhost
+        Listen 8080
+        ErrorLog "#{testpath}/httpd-error.log"
+        LogLevel warn
+        <IfModule log_config_module>
+        CustomLog "#{testpath}/httpd-access.log" common
+        </IfModule>
+        ServerRoot "#{testpath}"
```

Comment:
The basic config to get this running properly (with the sandbox):
```
        ServerRoot "#{opt_prefix}"
        PidFile "#{testpath}/httpd.pid"
        LoadModule authn_core_module modules/mod_authn_core.so
        LoadModule authz_core_module modules/mod_authz_core.so
        LoadModule unixd_module modules/mod_unixd.so
        LoadModule mpm_event_module modules/mod_mpm_event.so
        LoadModule dir_module modules/mod_dir.so
        DocumentRoot "#{testpath}"
        <Directory "#{testpath}">
        Require all granted
        </Directory>
```

Reply:
@kabel I've pushed that and it doesn't seem to work. Will let you look at logs when CI is done.

Reply:
The modules are install in to `/usr/local/opt/apache-httpd/libexec` so there's no modules directory. `s/modules/libexec/` and it should work.

Reply:
Yes, @javian is correct. Those paths I gave were for before you change the `libexec` configure option.